gateway/api/ray.py

```diff
@@ -76,17 +77,48 @@ def submit(self, job: Job) -> Optional[str]:
         """
         tracer = trace.get_tracer("scheduler.tracer")
         with tracer.start_as_current_span("submit.job") as span:
+            # get program
             program = job.program
+
+            # get dependencies
             _, dependencies = try_json_loads(program.dependencies)
-            with tarfile.open(program.artifact.path) as file:
-                extract_folder = os.path.join(
-                    sanitize_file_path(str(settings.MEDIA_ROOT)),
-                    "tmp",
-                    str(uuid.uuid4()),
+
+            # get artifact
+            working_directory_for_upload = os.path.join(
+                sanitize_file_path(str(settings.MEDIA_ROOT)),
+                "tmp",
+                str(uuid.uuid4()),
+            )
+            if program.image is not None:
+                # load default artifact
+                os.makedirs(working_directory_for_upload, exist_ok=True)
+                default_entrypoint_template = get_template("main.tmpl")
+                default_entrypoint_content = default_entrypoint_template.render(
+                    {
+                        "mount_path": settings.CUSTOM_IMAGE_PACKAGE_PATH,
+                        "package_name": settings.CUSTOM_IMAGE_PACKAGE_NAME,
+                    }
+                )
+                with open(
+                    os.path.join(
+                        working_directory_for_upload, DEFAULT_PROGRAM_ENTRYPOINT
+                    ),
+                    "w",
+                    encoding="utf-8",
+                ) as entrypoint_file:
+                    entrypoint_file.write(default_entrypoint_content)
+            elif bool(program.artifact):
```

Comment:
Curious about, can not we use `is not None` like in `image` here?

Reply:
`program.artifact` always return `FileInstance` :) so it is never `None`